tests/contrib/webapp2/test_webapp2.py

```diff
@@ -0,0 +1,132 @@
+import json
+
+import webapp2
+
+from ddtrace.constants import SAMPLING_PRIORITY_KEY
+from ddtrace.contrib.webapp2 import Webapp2TraceMiddleware
+from ddtrace.ext import errors, http
+from ...base import BaseTracerTestCase
+
+
+class MyHandler(webapp2.RequestHandler):
+    def post(self):
+        json_body = self.request.json
+        payload = json.dumps(json_body)
+        self.response.write(payload)
+
+
+class Webapp2TestCase(BaseTracerTestCase):
+    def setUp(self):
+        super(Webapp2TestCase, self).setUp()
+        handler = MyHandler
+        route = webapp2.Route('/tests/', handler, methods=['POST'])
+        route1 = webapp2.Route('/unused/', handler, methods=['POST'])
+        routes = [route, route1]
+        self.app = webapp2.WSGIApplication(routes=routes)
+        self.traced_app = Webapp2TraceMiddleware(
+            self.app, self.tracer, service='test-service')
+
+    def test_apm_middleware_valid_post(self):
+        # Given an application instrumented with a tracer which captures spans
+        # When doing a basic POST request
+        data = {'valid': 'test'}
+        payload = json.dumps(data)
+        request = webapp2.Request.blank('/tests/?q=example', POST=payload)
+        response = request.get_response(self.traced_app)
+
+        # Then the response will be successful
+        assert response.status_code == 200
+        assert response.text == payload
+        # And the tracer will contain a span with the correct tags for
+        # a POST request
+        span, = self.tracer.writer.pop()
+        assert span.span_type == 'http'
+        assert span.service == 'test-service'
+        assert span.resource == 'tests.contrib.webapp2.test_webapp2.MyHandler'
+        assert span.error == 0
+        assert span.get_tag(http.URL) == 'http://localhost/tests/?q=example'
```

Comment:
This should not contain query string if I'm not mistaken. (see the other frameworks).

Reply:
Thanks. Fixed this as well now.